nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1324 @@
+        if (
+            self._params.data_simulator.session_params.turn_prob < 0
+            or self._params.data_simulator.session_params.turn_prob > 1
+        ):
+            raise Exception("Turn probability is outside of [0,1]")
+        if (
+            self._params.data_simulator.session_params.overlap_prob < 0
+            or self._params.data_simulator.session_params.overlap_prob > 1
+        ):
+            raise Exception("Overlap probability is outside of [0,1]")
+
+        if (
+            self._params.data_simulator.session_params.mean_overlap < 0
+            or self._params.data_simulator.session_params.mean_overlap > 1
+        ):
+            raise Exception("Mean overlap is outside of [0,1]")
+        if (
+            self._params.data_simulator.session_params.mean_silence < 0
+            or self._params.data_simulator.session_params.mean_silence > 1
+        ):
+            raise Exception("Mean silence is outside of [0,1]")
+        if (
+            self._params.data_simulator.session_params.min_dominance < 0
+            or self._params.data_simulator.session_params.min_dominance > 1
+        ):
+            raise Exception("Minimum dominance is outside of [0,1]")
+        if (
+            self._params.data_simulator.speaker_enforcement.enforce_time[0] < 0
+            or self._params.data_simulator.speaker_enforcement.enforce_time[0] > 1
+        ):
+            raise Exception("Speaker enforcement start is outside of [0,1]")
+        if (
+            self._params.data_simulator.speaker_enforcement.enforce_time[1] < 0
+            or self._params.data_simulator.speaker_enforcement.enforce_time[1] > 1
+        ):
+            raise Exception("Speaker enforcement end is outside of [0,1]")
+
+        if (
+            self._params.data_simulator.session_params.min_dominance
+            * self._params.data_simulator.session_config.num_speakers
+            > 1
+        ):
+            raise Exception("Number of speakers times minimum dominance is greater than 1")
+
+        if (
+            self._params.data_simulator.session_params.overlap_prob
+            / self._params.data_simulator.session_params.turn_prob
+            > 1
+        ):
+            raise Exception("Overlap probability / turn probability is greater than 1")
+        if (
+            self._params.data_simulator.session_params.overlap_prob
+            / self._params.data_simulator.session_params.turn_prob
+            == 1
+            and self._params.data_simulator.session_params.mean_silence > 0
+        ):
+            raise Exception("Overlap probability / turn probability is equal to 1 and mean silence is greater than 0")
+
+        if (
+            self._params.data_simulator.session_params.window_type not in ['hamming', 'hann', 'cosine']
+            and self._params.data_simulator.session_params.window_type != None
+        ):
+            raise Exception("Incorrect window type provided")
+
+    def _get_speaker_ids(self):
```

Comment:
This would work as well, but it would also require looping over the manifest file entries to identify uniq speaker IDs before using random.choice. I can change this if that is preferable